backend/src/packages/auth/auth.controller.ts

```diff
@@ -146,6 +176,37 @@ class AuthController extends Controller {
       payload: options.user,
     };
   }
+
+  private async sendEmailResetPasswordLink(
+    options: ApiHandlerOptions<{
+      body: AuthRequestPasswordDto;
+    }>,
+  ): Promise<ApiHandlerResponse> {
+    const { origin, body } = options;
+
+    if (!origin) {
+      throw new HttpError({
+        message: 'Unspecified request origin!',
+        status: HttpCode.BAD_REQUEST,
+      });
+    }
+    const url = origin as string;
```

Comment:
I guess it would be better to cast `origin`, on the line 83. The same way as it was done for `body`
```js
  this.resetPassword(
    options as ApiHandlerOptions<{ body: AuthResetPasswordDto, origin: string }>,
  ),
```

Reply:
If I cast it there then in handler I will get origin = "undefined" in case of unspecified origin and I don't think it's a right way to do it.